server/api/study.py

```diff
@@ -300,31 +293,96 @@ def createStudy(self, params):
+        if len(set(params['userIds'])) != len(params['userIds']):
+            raise RestException('Duplicate user IDs.')
+        annotatorUsers = [
+            User.load(userId, user=creatorUser, level=AccessType.READ, exc=True)
+            for userId in params['userIds']
+        ]
+        duplicateAnnotations = Annotation.find({
+            'meta.studyId': study['_id'],
+            'meta.userId': {'$in': [
+                annotatorUser['_id'] for annotatorUser in annotatorUsers]}
+        })
+        if duplicateAnnotations.count():
+            # Just list the first duplicate
+            duplicateAnnotation = next(iter(duplicateAnnotations))
+            raise ValidationException(
+                'Annotator user "%s" is already part of the study.' %
+                duplicateAnnotation['meta']['userId'])
+        # Look up images only once for efficiency
+        images = Study.getImages(study)
+        for annotatorUser in annotatorUsers:
+            Study.addAnnotator(study, annotatorUser, creatorUser, images)
+
+        return self.getStudy(id=study['_id'], params={})
 
-        Study.addAnnotator(study, annotatorUser, creatorUser)
-        # TODO: return?
+    @describeRoute(
+        Description('Add images to a study.')
+        .param('id', 'The ID of the study.', paramType='path')
+        .param('imageIds', 'The image IDs to add, as a JSON array.',
+               paramType='form')
+        .errorResponse('ID was invalid.')
+        .errorResponse('You don\'t have permission to add a study image.', 403)
+    )
+    @access.user
+    @loadmodel(model='study', plugin='isic_archive', level=AccessType.READ)
+    def addImages(self, study, params):
+        Annotation = self.model('annotation', 'isic_archive')
+        Image = self.model('image', 'isic_archive')
+        Study = self.model('study', 'isic_archive')
+        User = self.model('user', 'isic_archive')
+
+        params = self._decodeParams(params)
+        self.requireParams('imageIds', params)
+
+        creatorUser = self.getCurrentUser()
+        User.requireAdminStudy(creatorUser)
+
+        # Load all images before adding any, to ensure all are valid and
+        # accessible
+        if len(set(params['imageIds'])) != len(params['imageIds']):
+            raise RestException('Duplicate image IDs.')
+        images = [
+            Image.load(
+                imageId, user=creatorUser, level=AccessType.READ, exc=True)
+            for imageId in params['imageIds']
+        ]
+        duplicateAnnotations = Annotation.find({
+            'meta.studyId': study['_id'],
+            'meta.imageId': {'$in': [image['_id'] for image in images]}
+        })
+        if duplicateAnnotations.count():
+            # Just list the first duplicate
+            duplicateAnnotation = next(iter(duplicateAnnotations))
+            raise ValidationException(
+                'Image "%s" is already part of the study.' %
+                duplicateAnnotation['meta']['imageId'])
```

Comment:
I was testing and found I can call this endpoint repeated with the same user ID without triggering this warning. I'd have to actually annotate an image before I'd get this warning, it looks like?